services/microbadger/microbadger-size.tester.js

```diff
@@ -0,0 +1,48 @@
+'use strict'
+
+const { isFileSize } = require('../test-validators')
+const t = (module.exports = require('../tester').createServiceTester())
+
+t.create('image size without a specified tag')
+  .get('/fedora/apache.json')
+  .expectBadge({
+    label: 'image size',
+    message: isFileSize,
+  })
+
+t.create('image size with a specified tag')
+  .get('/fedora/apache/latest.json')
+  .expectBadge({
+    label: 'image size',
+    message: isFileSize,
+  })
+
+t.create('missing download size')
+  .get('/puppet/puppetserver.json')
+  .intercept(nock =>
+    nock('https://api.microbadger.com')
+      .get('/v1/images/puppet/puppetserver')
+      .reply(200, {})
+  )
+  .expectBadge({ label: 'image size', message: 'invalid response data' })
+
+t.create('specified tag when repository has only one')
+  .get('/_/alpine/wrong-tag.json')
+  .expectBadge({ label: 'image size', message: 'not found' })
+
+t.create('nonexistent repository')
+  .get('/_/not-a-real-repo.json')
```

Comment:
I've switched the 'not found' tests to hit the live service, but previously we were mocking the 404. Were we mocking the 404 for a reason?

Reply:
Maybe for speed? The Microbadger tests have always been rather slow, we've had quite a few timeout issues in the past. But if the 404 response comes back quickly, I'm also happy for us to hit the live service. 👍

Reply:
I've not seen it time out, but we'll see how it goes..